djsonb/lookups.py

```diff
@@ -72,10 +74,27 @@ def sql(self):
                         sql_tuple = FilterTree.text_similarity_filter(rule[0], pattern, True)
                     else:
                         sql_tuple = FilterTree.text_similarity_filter(rule[0], pattern, False)
-                    rule_specs.append(sql_tuple)
+                    # add to the list of rules generated for this pattern (one per field)
+                    got_pattern = patterns.get(pattern) or []
+                    got_pattern.append(sql_tuple)
+                    patterns[pattern] = got_pattern
```

Comment:
This is an occasion for one of my favorite python tricks, though it might not be considered a change in the direction of readability: these three lines can be reduced to `patterns.setdefault(pattern, []).append(sql_tuple)`